### PR TITLE
Transfer credits now add up to total credits on sidebar

### PR DESCRIPTION
**Pull request by hellolol2016**
# Description

Shireen and I added the function to calculate the total amount of transfer credit on both types of sidebars (major and non-major). We changed the parameters to take in transferCourses on the non-major sidebar. The function calculates the sum and adds it to the totalCreditsInSchedule to get creditsTaken. 

For the future, maybe add AP courses on the "Add transfer courses" modal so the user doesn't have to look up the AP to NEU course transfer policies.

image

Closes # (issue number)

## Type of change

Please tick the boxes that best match your changes.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`
- [ ] This has migration changes and requires a run of `yarn dev:migration:run`

# Checklist:

- [ ] I have run the production builds in docker for the frontend/backend and ensure things run fine. Check README of repo on how to run if not sure.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where needed
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I've run the end to end tests
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ GraduateNU aims to empower Northeastern students to customize their plan of stud
 
 5. Then run the new version of the application by running `yarn dev` at the root of the project. This starts up a NestJS server + a NextJS frontend + a Proxy. The proxy listens on port [3002](http://localhost:3002/), forwards /api requests to the NestJS server running on port 3001, and all other requests to the frontend running on port 3000.
 
+5a. For Windows machines, run `yarn concurrently "yarn workspaces foreach --parallel --verbose --interlaced run dev" "yarn dev:proxy"` instead of `yarn dev`
```
Comment (KobeZ123) [inline README.md:17]: thanks for adding this!

File: packages/frontend/components/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -247,12 +253,19 @@ const Sidebar: React.FC<SidebarProps> = memo(
...
   selectedPlan: PlanModel<string>;
+  transferCourses: ScheduleCourse2<unknown>[];
 }
 
 export const NoMajorSidebar: React.FC<NoMajorSidebarProps> = ({
   selectedPlan,
+  transferCourses,
 }) => {
-  const creditsTaken = totalCreditsInSchedule(selectedPlan.schedule);
+  const transferCredits = transferCourses.reduce(
```
Comment (KobeZ123) [inline packages/frontend/components/Sidebar/Sidebar.tsx:263]: does the regular sidebar (the one that is not "no major") already support adding transfer course credits? if not, we should abstract the logic and add to both locations.

**Review by KobeZ123 — CHANGES_REQUESTED** [review]

File: packages/frontend/components/Sidebar/Sidebar.tsx (hunk 2 above)
Comment (hellolol2016) [reply packages/frontend/components/Sidebar/Sidebar.tsx:263]: Yes, I believe lines 193 - 199 add to the creditsTaken variable for the normal sidebar.

File: packages/frontend/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -247,12 +253,19 @@ const Sidebar: React.FC<SidebarProps> = memo(
...
+    (sum, course) => course.numCreditsMin + sum,
+    0
+  );
+  const creditsTaken =
```
Comment (KobeZ123) [inline packages/frontend/components/Sidebar/Sidebar.tsx:267]: can we move this logic calculating schedule + transfer courses into one function that returns a credit value?

**Review by KobeZ123 — CHANGES_REQUESTED** [review]

**Review by KobeZ123 — APPROVED** [review]
LGTM!

**Pull request merged**